### PR TITLE
feat(act): load real pretrained checkpoints and validate OpenVINO export parity

### PR DESCRIPTION
**Pull request by pwolnows**
## Summary

`ACT` already advertises OpenVINO/ONNX/ExecuTorch export support via `ExportablePolicyMixin.get_supported_export_backends()`, but there was no way to load a **real pretrained checkpoint** (e.g. a LeRobot `ACTPolicy.from_pretrained(...)` checkpoint) into physicalai's native `ACT` implementation — so the export path could only ever be exercised on randomly-initialized weights. This PR closes that gap and adds integration tests that prove the exported model reproduces the native model's behavior.

## Changes

- **`library/src/physicalai/policies/act/policy.py`**
  - `ACT.__init__` gains an optional `pretrained_name_or_path: str | Path | None` argument.
  - New `_from_hf(...)` resolves a local directory or HF Hub repo id (via `hf_hub_download`) for `config.json` + `model.safetensors`, and rebuilds the `ACTConfig` from the HF config fields.
  - New `_remap_lerobot_act_state_dict(...)` translates LeRobot checkpoint key prefixes (`model.`, `normalize_targets.buffer_action`, `normalize_inputs.buffer_observation_state`, `unnormalize_outputs.buffer_action`, plus per-camera image normalizer buffers) to physicalai's native parameter names.
  - `_initialize_model()` accepts an optional `weights_file`, loading + remapping the checkpoint's `safetensors` weights via `load_state_dict(strict=False)` (raises on genuinely unexpected keys, warns on missing keys).

- **`library/src/physicalai/policies/act/preprocessor.py`**
  - Fixed an argument-order bug: `image_resolution` is stored as `(height, width)`, but `_resize_with_ar_pad(img, width, height)` expects width first. This silently swapped dimensions for any non-square image resolution.

- **`library/tests/integration/test_act_openvino_parity.py`** (new)
  - Loads a real ACT checkpoint natively, exports it to OpenVINO, and validates parity two ways:
    1. **Numerical** (`TestACTOpenVINONumericalParity`): `predict_action_chunk` max-abs-diff and cosine similarity across 20 observations sampled from the `gym-aloha` env.
    2. **Closed-loop** (`TestACTOpenVINOClosedLoopParity`): per-episode success/fail outcomes and success-rate delta on `AlohaTransferCube-v0` with matching seeds.
  - Both test classes are marked `@pytest.mark.slow` and run on demand:
    ```
    pytest -m slow tests/integration/test_act_openvino_parity.py
    ```

## Validation results

Ran against the public [`lerobot/act_aloha_sim_transfer_cube_human`](https://huggingface.co/lerobot/act_aloha_sim_transfer_cube_human) checkpoint, 20 numeric samples + 10 closed-loop episodes:

- **Numerical**: native vs. OpenVINO-exported `predict_action_chunk` outputs differ by **~8e-7 max abs diff** (cosine similarity ≈ 1.0) — essentially floating-point noise. Well within the `0.05` tolerance.
- **Closed-loop**: identical **60% success rate** (6/10) on both backends, with **100% per-episode outcome match** — every one of the 10 seeds produced the same success/fail result on native and exported models, with near-identical episode step counts. Well within the `15pp` tolerance.

This gives concrete evidence that the OpenVINO export path is numerically and behaviorally faithful to the native model once loaded from a real checkpoint.

## Notes

- `compress_to_fp16` is set to `False` in the integration test; FP16 weight compression has previously been observed to cause accuracy regressions for ACT, so parity validation is done at FP32.
- Happy to adjust checkpoint remap key handling if there's a preferred convention for LeRobot state-dict compatibility.

### CODE REVIEW

File: scripts/compare_act_backends.py [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+    parser.add_argument("--n-numeric-samples", type=int, default=20, help="Number of single-step samples to compare")
+    parser.add_argument("--n-episodes", type=int, default=10, help="Number of closed-loop episodes to compare")
+    parser.add_argument("--n-action-steps", type=int, default=100, help="Action-chunk consumption length")
+    parser.add_argument("--seed-offset", type=int, default=0, help="Seed offset shared by both comparisons")
+    parser.add_argument("--seed", type=int, default=42, help="Global RNG seed for reproducibility")
+    parser.add_argument("--max-abs-diff-tolerance", type=float, default=0.05,
+                        help="Numeric comparison PASS threshold for max abs diff")
+    parser.add_argument("--success-rate-diff-tolerance", type=float, default=15.0,
+                        help="PASS threshold (percentage points) for |native_rate - exported_rate|")
+    parser.add_argument("--output", type=Path, default=Path("/tmp/compare_act_backends_report.json"),  # noqa: S108
```
[the remark this answers is not in the document]
Comment (pwolnows) [reply scripts/compare_act_backends.py]: resolved

File: scripts/compare_act_backends.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+        "per_episode_outcome_match_pct": matches / n_episodes * 100,
+    }
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument("--checkpoint", default="lerobot/act_aloha_sim_transfer_cube_human",
+                        help="HF Hub repo id or local dir with config.json + model.safetensors")
+    parser.add_argument("--export-dir", type=Path, default=Path("/tmp/act_export_openvino"),  # noqa: S108
```
[the remark this answers is not in the document]
Comment (pwolnows) [reply scripts/compare_act_backends.py]: resolved

File: scripts/compare_act_backends.py [hunk 3]
```diff
@@ -0,0 +1,401 @@
+# Copyright (C) 2026 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+r"""Compare native PyTorch ACT vs its OpenVINO export - numerical and closed-loop parity.
```
Comment (sovrasov) [inline scripts/compare_act_backends.py]: Can we perhaps move this to the test suit in the inference repo? It looks like the suite aims similar scenario: compare vs a source of golden results, so we don't need a standalone script here to cover this

Comment (pwolnows) [reply scripts/compare_act_backends.py]: moved to tests/integration/test_act_openvino_parity.py

**Review by sovrasov — APPROVED** [review]
